src/cpp/session/SessionConsoleProcess.cpp

```diff
@@ -301,14 +306,79 @@ bool ConsoleProcess::onContinue(core::system::ProcessOperations& ops)
    return true;
 }
 
+Error ConsoleProcess::getLogFile(FilePath& file) const
```

Comment:
We don't pass mutable values by reference, rather we pass them as naked pointers so the fact that mutation side effects are occurring is clear both in the function and at the call site. Passing a naked pointer is "safe" in our system b/c you don't have to reason about whether you need to delete it (you never do) or whether it's NULL (we don't allow NULL pointers -- rather we use empty smart pointers). See this for further rationale: https://google.github.io/styleguide/cppguide.html#Reference_Arguments